docs/guides/python/serverless-llama.mdx

```diff
@@ -0,0 +1,275 @@
+
+## What we'll be doing
+
+We will use the [Llama](https://huggingface.co/) models from Hugging Face for natural language processing, combined with Nitric to manage the API routes and storage.
+
+1. Setting up the environment.
+2. Creating the translation service.
+3. Deploying the service.
+4. Testing the translation functionality.
+
+## Prerequisites
+
+- [uv](https://docs.astral.sh/uv/#getting-started) - for Python dependency management
+- The [Nitric CLI](/get-started/installation)
+- _(optional)_ An [AWS](https://aws.amazon.com) account
+
+## Project setup
+
+We'll start by creating a new project for our translator service using Nitric's python starter template.
+
+```bash
+nitric new translator py-starter
+cd translator
+```
+
+Next, let's install our base dependencies, then add the extra dependencies we need specifically for loading our language model.
+
+```bash
+# Install the base dependencies
+uv sync
+uv add llama-cpp-python
+```
+
+You will also need to [download the Llama model](https://huggingface.co/bartowski/Llama-3.2-1B-Instruct-GGUF/tree/main) file and ensure it is located in the `./models/` directory with the correct model file name.
+
+In this guide we'll be using `Llama-3.2-1B-Instruct-Q4_K_M.gguf`, this model is ideal for serverless because its reduced size and efficient 4-bit quantization make it cost-effective and scalable, running smoothly within the resource limits of serverless compute environments while maintaining solid performance.
+
+Your folder structure should look like this:
+
+```bash
+/translator
+    /models
+        Llama-3.2-1B-Instruct-Q4_K_M.gguf
+    /services
+        api.py
+    nitric.yaml
+    pyproject.toml
+    python.dockerfile
+    python.dockerfile.ignore
+    README.md
+    uv.lock
+```
+
+## Creating the translation service
+
+Our project will use Nitric to handle API requests, and we will process the text translation using Llama. The results will be stored in a Nitric bucket.
+
+Let's start by defining the translation logic using the Llama model:
+
+Remove the contents of `services/api.py` and update it with the following code that will load the Llama model and implement the translation functionality. We'll also do some basic calculations for evaluation times for tokens:
+
+```python title:services/api.py
+from llama_cpp import Llama
+import time
+
+# Load the locally stored Llama model
+llama_model = Llama(model_path="./models/Llama-3.2-1B-Instruct-Q4_K_M.gguf")
+
+# Function to perform translation using the Llama model
+def translate_text(text):
+    prompt = f'Translate "{text}" to Spanish.'
+    
+    start_time = time.time()
+
+    # Generate a response using the locally stored model
+    response = llama_model(
+        prompt=prompt,
+        max_tokens=150,  
+        temperature=0.7,
+        top_p=0.9,
+        stop=["\n"]  
+    )
```

Comment:
Not sure if it's worthwhile, but making these options to show off a bit more configurability could be good.

e.g. 

```python
@main.post("/translate")
async def handle_translation(ctx: HttpContext):
    max_tokens = ctx.req.query.get("max_tokens", default_max_tokens)
    preset = ctx.req.query.get("temperature", default_temperature)
  
    text = ctx.req.json["text"]
```

We also support using raw text in the dashboard api testing. So not all prompts need to be wrapped in JSON